client/cli/src/params/pruning_params.rs

```diff
@@ -23,57 +23,78 @@ use sc_service::{BlocksPruning, PruningMode};
 /// Parameters to define the pruning mode
 #[derive(Debug, Clone, PartialEq, Args)]
 pub struct PruningParams {
-	/// Specify the state pruning mode, a number of blocks to keep or 'archive'.
+	/// Specify the state pruning mode.
 	///
-	/// Default is to keep only the last 256 blocks,
-	/// otherwise, the state can be kept for all of the blocks (i.e 'archive'),
-	/// or for all of the canonical blocks (i.e 'archive-canonical').
-	#[arg(alias = "pruning", long, value_name = "PRUNING_MODE")]
-	pub state_pruning: Option<String>,
-	/// Specify the blocks pruning mode, a number of blocks to keep or 'archive'.
+	/// This mode specifies when the block's state (ie, storage)
+	/// should be pruned (ie, removed) from the database.
+	#[arg(alias = "pruning", long, value_name = "PRUNING_MODE", default_value = "256")]
+	pub state_pruning: DatabasePruningMode,
```

Comment:
We need to tell clap that `DatabasePruningMode` is an enum value. So, that it shows the possible values you can pass to it. As I said in my initial comment, you probably need to implement this trait as the derive macro will maybe not deliver good results. However, you can first try the macro and look at the results :)